File: sdks/typescript/smoke.mjs
import { Memory } from './client.mjs'

const base = process.env.BARQ_BASE ?? 'http://127.0.0.1:18099'
const client = new Memory(base)

const saved = await client.remember('TypeScript SDK smoke fact', { tenantId: 'acme' })
if (!saved.id) throw new Error('remember failed')

const hits = await client.recall('sdk smoke fact', 'acme', 5)
if (!hits.some((h) => h.id === saved.id)) throw new Error('recall failed')

const successor = await client.update(saved.id, 'TypeScript SDK smoke fact v2')
const chain = await client.history(successor.id)
if (chain.length !== 2) throw new Error(`history ${chain.length} != 2`)

await client.forget(successor.id)
if ((await client.get(successor.id)) !== null) throw new Error('forgotten still visible')

console.log('TYPESCRIPT SDK SMOKE TEST OK')

File: sdks/python/smoke.py
"""End-to-end Python SDK smoke test against a running server."""
import os
import sys

sys.path.insert(0, os.path.dirname(__file__))
from barq_memory import Memory

base = os.environ.get("BARQ_BASE", "http://127.0.0.1:18099")
client = Memory(base)

saved = client.remember("Python SDK smoke fact", tenant_id="acme")
assert saved.id and saved.text == "Python SDK smoke fact"

hits = client.recall("sdk smoke fact", tenant_id="acme", limit=5)
assert any(h.id == saved.id for h in hits), "recall failed"

successor = client.update(saved.id, "Python SDK smoke fact v2")
chain = client.history(successor.id)
assert len(chain) == 2, f"history {len(chain)} != 2"

client.forget(successor.id)
assert client.get(successor.id) is None, "forgotten still visible"

print("PYTHON SDK SMOKE TEST OK")
